packages/sitecore-jss/src/graphql-request-client.ts

```diff
@@ -92,12 +92,13 @@ export class DefaultRetryStrategy implements RetryStrategy {
   private factor: number;
 
   /**
-   * @param {number[]} statusCodes HTTP status codes to trigger retries on
-   * @param {number} factor Factor by which the delay increases with each retry attempt
+   * @param {Object} options Configurable options for retry mechanism.
+   * @param {number[]} options.statusCodes HTTP status codes to trigger retries on
+   * @param {number} options.factor Factor by which the delay increases with each retry attempt
    */
-  constructor(statusCodes?: number[], factor?: number) {
-    this.statusCodes = statusCodes || [429];
-    this.factor = factor || 2;
+  constructor(options: { statusCodes?: number[]; factor?: number }) {
```

Comment:
Need a default empty object to make `options` optional. Probably good to add a `DefaultRetryStrategy` unit test to cover that. e.g. 'should use statusCode 429 and factor 2 by default'
```suggestion
  constructor(options: { statusCodes?: number[]; factor?: number } = {}) {
```